glow/src/backend.rs

```diff
@@ -30,8 +31,60 @@ impl Backend {
             settings.text_multithreading,
         );
 
-        let quad_pipeline = quad::Pipeline::new(gl);
-        let triangle_pipeline = triangle::Pipeline::new(gl);
+        let version = gl.version();
+        let shader_version = match (
+            version.major,
+            version.minor,
+            version.is_embedded,
+        ) {
+            // OpenGL 3.0+
+            (3, 0 | 1 | 2, false) => (
+                format!("#version 1{}0", version.minor + 3),
+                format!(
+                    "#version 1{}0\n#define HIGHER_THAN_300 1",
+                    version.minor + 3
+                ),
+            ),
```

Comment:
Maybe it could be a part of the `program` module?

Reply:
I've added a `Version` struct similar to what you described and further separated a new `Shader` struct from the `program::create`. Unfortunately, I can't simplify the `include_str!` since it doesn't accept `&'static str` (only string literals).

Check 424e1d3 and let me know what you think!

Reply:
Looks awesome! :ok_hand: